### PR TITLE
Connection class added

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | https://github.com/yiisoft/yii2/pull/18560

This fixes regression introduced in yiisoft/yii2#18476
Tests are expected to fail.

### CODE REVIEW

File: tests/ActiveDataProviderTest.php [hunk 1]
```diff
@@ -50,13 +50,15 @@ public function testQuery()
         $query->from('customer');
 
         $provider = new ActiveDataProvider([
+            'connectionClass' => 'yii\mongodb\Connection',
             'query' => $query,
             'db' => $this->getConnection(),
```
Comment (samdark) [inline tests/ActiveDataProviderTest.php:55]: Can't we determine connection from what's in db if it's specified?

Comment (bizley) [reply tests/ActiveDataProviderTest.php:55]: You mean like `get_class()`? What is the purpose?

Comment (samdark) [reply tests/ActiveDataProviderTest.php:55]: Not to specify a class if we have a class already?

Comment (bizley) [reply tests/ActiveDataProviderTest.php:55]: If I understand correctly - you would like `yii\data\ActiveDataProvider` to skip `ensure` checks on object-like `db`? Still strings and arrays need to be checked.

**Comment by bizley** [comment]
Closing since we are not adding `connectionClass`.

**Pull request closed without merging**